vscode-extension/src/utilities/versionUpdateUtils.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Utils file for handling actions to perform whenever user
+ * installs or updates the extension.
+ * VS Code automatically checks for version updates
+ * (https://code.visualstudio.com/docs/editor/extension-marketplace#_extension-autoupdate)
+ * but I haven't been able to find a way to use this to trigger any
+ * actions. Also even though the Walkthrough documentation says
+ * that this gets triggered on first install, I haven't been able
+ * to trigger this either, so I am calling this manually
+ * https://code.visualstudio.com/api/references/contribution-points#contributes.walkthroughs
+ */
+
+import * as vscode from "vscode";
+
+import { EXTENSION_NAME } from "../util";
+import { VERSION_KEY_NAME } from "../constants";
+
+export async function performVersionInstallAndUpdateActionsIfNeeded(
+  context: vscode.ExtensionContext
+) {
+  const extension = vscode.extensions.getExtension(context.extension.id);
+  const currExtensionVersion = extension?.packageJSON.version;
+  const config = vscode.workspace.getConfiguration(EXTENSION_NAME);
+  const lastActivatedVersion = config.get<string>(VERSION_KEY_NAME);
+  await config.update(
+    VERSION_KEY_NAME,
+    currExtensionVersion,
+    vscode.ConfigurationTarget.Global
+  );
```

Comment:
I think this should only be done if it's not already set / doesn't match. Otherwise it'll unnecessarily write the same version each time

Reply:
Yea sure I can do that